playbook-base.yaml

```diff
@@ -1,76 +1,160 @@
-- name: Basic host setup
-  hosts: workers
+# Base playbook sets up a Kubernetes cluster using kubeadm and kine.
+# This playbook is idempotent, so it can be used on existing clusters to add new nodes, or sync their state.
+# This playbook deploys a multi-node setup. It is written with a future HA deployment in mind, but for now it assumes
+# a single CP node.
+
+# The following values are hadrcoded across the playbook:
+# - 10.0.0.8 as the api-server IP
+# - 10.0.0.9 as the kine (etcd) IP
+# Mainly because Ctrl-F is faster than what it would take me to use variables.
+# Maybe in the future these IPs will be load-balanced for HA. For now, they are just extra addresses in the CP node.
+
+- name: Common node configuration
+  hosts: control-plane:workers
   remote_user: root
   strategy: free
   tasks:
-  - name: Core packages are installed
-    community.general.pacman:
-      # https://unix.stackexchange.com/questions/274727/how-to-force-pacman-to-answer-yes-to-all-questions/584001#584001
-      extra_args: --ask 4 --noconfirm
-      state: present
-      name: [
-        # Container runtime
-        cri-o, crictl, crun,
-        # Kubelet & kubectl
-        kubelet, kubectl,
-        # Kubeadm is needed in worker nodes as it provides /usr/lib/systemd/system/kubelet.service.d/10-kubeadm.conf
-        kubeadm,
-        # NFS
-        nfs-utils,
-        # Misc node administration tools.
-        btop, htop, lsof, nano
-      ]
+    - name: Core packages are installed
+      community.general.pacman:
+        # https://unix.stackexchange.com/questions/274727/how-to-force-pacman-to-answer-yes-to-all-questions/584001#584001
+        extra_args: --ask 4 --noconfirm
+        state: present
+        name: [
+          # Container runtime
+          cri-o, crictl, crun,
+          # Kubelet & kubectl
+          kubelet, kubectl,
+          # Kubeadm is needed in worker nodes as it provides /usr/lib/systemd/system/kubelet.service.d/10-kubeadm.conf
+          kubeadm,
+          # NFS
+          nfs-utils,
+          # Misc node administration tools.
+          btop, htop, lsof, nano
+        ]
 
-  - name: Copy folder structure
-    ansible.builtin.file:
-      path: "/{{ item.path }}"
-      state: directory
-      recurse: yes
-    with_community.general.filetree: "{{ playbook_dir }}/files/"
-    loop_control:
-      label: "/{{ item.path }}"
-    when: item.state == 'directory'
+    - name: Copy folder structure
+      ansible.builtin.file:
+        path: "/{{ item.path }}"
+        state: directory
+        recurse: yes
```

Comment:
I would not recurse it as it would copy files inside folders and you just want to have the folder in the same way you have it in the computer you are running Ansible.

Files are going to be copied or templated above. Or am I losing something here?

Reply:
Is `file`, not `copy`. I do not know what does file's recurse. Ignore this comment for now.

Reply:
Instead of `recurse` that could modify all child folders, I would add something like this:
 ```yaml
 mode: "{{ item.mode }}"
 user: "{{ item.owner }}"
 group: "{{ item.group }}"
 ```
 so you can manage the ownership of the target based on ownerships of the tree you have locally.
 
 But now this is a NIT.

Reply:
I though about this, but so far I haven't found an use case for creating files with anything other than the default umask.